prow/cmd/pipeline/controller.go

```diff
@@ -192,6 +192,14 @@ func newController(opts controllerOptions) (*controller, error) {
 	return c, nil
 }
 
+// clusterToCtx converts the prow job's cluster to a cluster context
+func clusterToCtx(cluster string) string {
+	if cluster == kube.DefaultClusterAlias {
+		return kube.InClusterContext
```

Comment:
This is not always true

Reply:
Which is the use case when this is not true?

Reply:
It seems that there is an issue when both `DefaultClusterAlias` and `InClusterContext` point to the same cluster (this is the default). The PipelineRun resource created for a prow job with `spec.Cluster: default`, is removed by the `InClusterContext` reconciliation because `spec.Cluster: default != spec.Cluster: ""`

Reply:
I explained the issue here for build controller https://github.com/kubernetes/test-infra/pull/12177#discussion_r275317270

Reply:
This seems more appropriate for the other PR. I'd recommend taking out the commit from here as I don't think it works correctly. But happy to address it in a follow-up

Reply:
I removed that commit from this PR. We'll work out a solution in a follow up PR. Thanks